rdsbroker/broker.go

```diff
@@ -209,9 +215,35 @@ func (b *RDSBroker) Provision(
 		if err != nil {
 			return brokerapi.ProvisionedServiceSpec{}, err
 		}
+
+		if provisionParameters.RestoreFromLatestSnapshotBefore != nil {
+			if *provisionParameters.RestoreFromLatestSnapshotBefore == "" {
+				return brokerapi.ProvisionedServiceSpec{}, fmt.Errorf("Parameter restore_from_latest_snapshot_before must not be empty")
+			}
+
+			restoreFromLatestSnapshotBeforeTime, err := time.ParseInLocation(
+				RestoreFromLatestSnapshotBeforeTimeFormat,
+				*provisionParameters.RestoreFromLatestSnapshotBefore,
+				time.UTC,
+			)
+			if err != nil {
+				return brokerapi.ProvisionedServiceSpec{},
+					fmt.Errorf("Parameter restore_from_latest_snapshot_before should be a date and a time: %s", err)
+			}
+
+			prunedSnapshots := make([]*rds.DBSnapshot, 0)
+			for _, snapshot := range snapshots {
+				if snapshot.SnapshotCreateTime.Before(restoreFromLatestSnapshotBeforeTime) {
+					prunedSnapshots = append(prunedSnapshots, snapshot)
+				}
+			}
+			snapshots = prunedSnapshots
+		}
+
 		if len(snapshots) == 0 {
 			return brokerapi.ProvisionedServiceSpec{}, fmt.Errorf("No snapshots found for guid '%s'", *provisionParameters.RestoreFromLatestSnapshotOf)
 		}
+
 		snapshot := snapshots[0]
```

Comment:
Could we add a log line show which snapshot is being used, since now it isn't necessarily the one most recent to the time of the other log lines?

Reply:
This is captured in a tag on the subsequently created RDS instance: `Restored From Snapshot`

Reply:
What about in the event that database creation fails? I think if I was debugging that, I'd like to be able see in the logs which snapshot it picked

Reply:
Good shout, added in e4f482944eb2e1bd320b8a5577c242a5389242ea